src/CodeCoverageToLcovConverter/CodeCoverageReader.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Xml.Linq;
+
+namespace CodeCoverageToLcovConverter
+{
+    public class CodeCoverageReader
+    {
+        public CodeCoverageDataModel ParseCoverageFile(XDocument coverageFileContents, XNamespace nameSpace)
+        {
+            if (coverageFileContents == null)
+            {
+                throw new ArgumentNullException(nameof(coverageFileContents));
+            }
+
+            if (coverageFileContents.Root == null)
+            {
+                throw new ArgumentException("Expected valid .coverage xml data.");
+            }
+
+            var xmlCodeCoverageData = new CodeCoverageDataModel();
+
+            var modules = coverageFileContents.Elements(nameSpace + "results").FirstOrDefault().Elements(nameSpace + "modules").FirstOrDefault().Elements(nameSpace + "module");
+
+            foreach(var module in modules)
+            {
+                var sourceFiles = module.Elements(nameSpace + "source_files").Elements(nameSpace + "source_file");
+
+                foreach (var sourceFile in sourceFiles)
+                {
+                    xmlCodeCoverageData.Sources.Add(sourceFile.Attribute(nameSpace + "path").Value);
+                }
+
+                var functions = module.Elements(nameSpace + "functions").FirstOrDefault().Elements(nameSpace + "function");
+
+                int sourcesAddedUpUntillTheLastModule = 0;
```

Comment:
We need a better name for this. We should ideally refine this logic.
Have a separate model which exactly maps coverage xml.